### PR TITLE
Sdk: expose bridge tx status

### PR DESCRIPTION
**Pull request by ChiTimesChi**
**Description**
Added two functions to track the Synapse transactions:
- `getBridgeID(originChainId, bridgeModuleName, txHash)` returns `bridgeID`, which can be used to track the transaction status.
  - If the provided transaction is not an "origin transaction" for the corresponding "bridge module" (SynapseBridge / SynapseCCTP), this will throw an explicit error: 7f92da2
 - `getBridgeTxStatus(destChainId, bridgeModuleName, bridgeID)` returns the status of the transaction on the destination chain.
   - This will always return `false` for unrelated `bridgeID` 

**Additional context**
Add any other context about the problem you're solving.

**Metadata**
- Fixes #[Link to Issue]

## Summary by CodeRabbit

- **New Features**
	- Added bridge transaction ID retrieval and transaction completion status check functions.

- **Enhancements**
	- Improved estimation of bridge operation times for enhanced user experience.

- **Tests**
	- Added tests to ensure reliability of new bridge transaction features.

- **Documentation**
	- Updated public API documentation to include new bridge transaction tracking functions.

### CODE REVIEW

**Comment by ChiTimesChi** [comment]
Ready for the initial review. Open questions so far:
- Is `bridgeID` a good naming? The idea is to abstract away the difference between's SynapseBridge `kappa` and SynapseCCTP `requestID` from the SDK consumer.
- Should `getBridgeID()` for SynapseBridge throw an error, if the provided transaction does not represent a valid origin SynapseBridge transaction?
  - Right now the behavior is that `bridgeID` is returned which always yields status `false` in `getBridgeTxStatus()`
  - Could throw an error to be consistent with SynapseCCTP behavior, but that would require to do one extra `getTransactionReceipt` RPC call.

**Comment by ChiTimesChi** [comment]
The previous [comment](https://github.com/synapsecns/sanguine/pull/1658#issuecomment-1855780618) was partly resolved in
- cd19b3b

**Comment by Defi-Moses** [comment]
`bridgeID` works but doesn't intuitively tell me that its asking for the unique transaction identifier. I think something along the lines of `transactionID` or `synapseID`. I like something along the lines of `synapseTxId` because its clear but might be a little bit wordy.

**Comment by bigboydiamonds** [comment]
> `bridgeID` works but doesn't intuitively tell me that its asking for the unique transaction identifier. I think something along the lines of `transactionID` or `synapseID`. I like something along the lines of `synapseTxId` because its clear but might be a little bit wordy.

Agreed that `bridgeID` _could_ be confusing whether that refers to `transactionHash` or `kappa` when calling SDK from frontend

Adding a few suggestions for consideration:

`getBridgeKappa()` => returns `bridgeKappa`
`getBridgeTxnKappa()` => returns `bridgeTxnKappa`

**Review by trajan0x — APPROVED** [review]

**Comment by ChiTimesChi** [comment]
`kappa` is specific to the SynapseBridge, which is only one of the solutions that SDK is supposed to abstract away from the end consumer, so it doesn't feel too nice.

I like `synapseTxId` over `transactionId`, as the latter could be easily confused with `transactionHash`, while the former is quite explicit.

File: packages/sdk-router/src/router/synapseRouterSet.ts [hunk 1]
```diff
@@ -36,6 +36,26 @@ export class SynapseRouterSet extends RouterSet {
...
+    return this.getSynapseRouter(originChainId).getSynapseTxId(txHash)
+  }
+
+  /**
+   * @inheritdoc RouterSet.getBridgeTxStatus
+   */
+  public async getBridgeTxStatus(
+    destChainId: number,
+    synapseTxId: string
+  ): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (ChiTimesChi) [reply packages/sdk-router/src/router/synapseRouterSet.ts:55]: Nice hallucination bruh

[the remark this answers is not in the document]
Comment (ChiTimesChi) [reply packages/sdk-router/src/router/synapseRouterSet.ts:55]: I would consider giving you a shot-out once you learn how to use the markdown links properly

**Pull request merged**